File: adapters/stdout.py
#!/usr/bin/env python3
"""Stdout adapter — agent-agnostic plain text output.

Any agent that can run a shell command and read stdout can use this.
Reads prompt from stdin (raw text or {"prompt": "..."} JSON).
Exits 0 with no output when no task signal is detected.
Exits 0 with plain directive text when skills are relevant.

Example (Vanta hook, generic CI, any shell-invocable agent):
  prompt=$(cat) && echo "$prompt" | python3 /path/to/adapters/stdout.py
"""
import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent))
from core.classify import classify, FALLBACK_HINT

raw = sys.stdin.read()
try:
    prompt = json.loads(raw).get("prompt") or ""
except (ValueError, AttributeError):
    prompt = raw

results = classify(prompt)
if not results:
    sys.exit(0)

if results == [("_fallback", FALLBACK_HINT)]:
    print(FALLBACK_HINT)
else:
    rows = "; ".join(f"{label} -> {skills}" for label, skills in results)
    print(
        "Relevant skills detected. Before acting invoke the matching skills "
        "(process skills like brainstorming/debugging FIRST, then implementation). "
        + rows
    )

File: src/adapters/stdout.ts
#!/usr/bin/env node
/** Stdout adapter — plain text, any agent that reads shell output.
 *
 * Reads prompt from stdin (raw text or {"prompt":"..."} JSON).
 * Silent (exit 0, no output) when no task signal detected.
 */
import { classify, FALLBACK } from "../classify.js";

const raw = await new Promise<string>((res) => {
  let buf = "";
  process.stdin.setEncoding("utf8");
  process.stdin.on("data", (d) => (buf += d));
  process.stdin.on("end", () => res(buf));
});

let prompt = "";
try {
  const parsed = JSON.parse(raw);
  prompt = parsed?.prompt ?? "";
} catch {
  prompt = raw;
}

const results = classify(prompt);
if (!results.length) process.exit(0);

const isFallback = results.length === 1 && results[0]!.label === "_fallback";
if (isFallback) {
  process.stdout.write(FALLBACK.skills + "\n");
} else {
  process.stdout.write(
    "Relevant skills detected. Before acting invoke the matching skills " +
    "(process skills like brainstorming/debugging FIRST, then implementation). " +
    results.map((r) => `${r.label} -> ${r.skills}`).join("; ") + "\n"
  );
}
